### PR TITLE
feat: implement testPermission api of organizations

### PR DESCRIPTION
**Pull request by athakor**
Towards #89

### CODE REVIEW

**Comment by athakor** [comment]
@chingor13 gentale ping

File: src/main/java/com/google/cloud/resourcemanager/ResourceManager.java [hunk 1]
```diff
@@ -337,4 +337,20 @@ public static ProjectListOption fields(ProjectField... fields) {
...
    */
   List<Boolean> testPermissions(String projectId, List<String> permissions);
+
+  /**
+   * Returns a list of booleans representing whether the caller has the permissions on the specified
+   * Organization.
+   *
+   * @param resource the organization's resource name, e.g. "organizations/123"
+   * @param permissions the set of permissions to check for the resource. Permissions with wildcards
+   *     (such as '*' or 'storage.*') are not allowed
```
Comment (elharo) [inline src/main/java/com/google/cloud/resourcemanager/ResourceManager.java:347]: period

File: src/main/java/com/google/cloud/resourcemanager/ResourceManager.java [hunk 2]
```diff
@@ -337,4 +337,20 @@ public static ProjectListOption fields(ProjectField... fields) {
...
+   * @return A list of booleans representing whether the caller has the permissions specified (in
```
Comment (elharo) [inline src/main/java/com/google/cloud/resourcemanager/ResourceManager.java:348]: nit: A --> a

File: src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java [hunk 3]
```diff
@@ -301,4 +301,25 @@ public Policy replacePolicy(String projectId, Policy newPolicy) throws ResourceM
       throw translate(ex);
     }
   }
+
+  @Override
+  public List<Boolean> testOrgPermissions(String resource, List<String> permissions) {
```
Comment (elharo) [inline src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java:306]: This API is strange. One rarely encounters a list of booleans. What does someone do with it? How do you know which boolean maps to which permission?

File: src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/ResourceManagerRpc.java [hunk 4]
```diff
@@ -124,5 +124,12 @@ Integer getInt(Map<Option, ?> options) {
    */
   List<Boolean> testPermissions(String projectId, List<String> permissions);
 
-  // TODO(ajaykannan): implement "Organization" functionality when available (issue #319)
+  /**
+   * Tests whether the caller has the given permissions on the specified Organization. Returns a
+   * list of booleans corresponding to whether or not the user has the permission in the same
```
Comment (elharo) [inline src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/ResourceManagerRpc.java:129]: Do we maybe just want the logical AND of all these permissions?

Or perhaps what you want to return is a Map<String, boolean> that lists the permissions and their results?

File: src/test/java/com/google/cloud/resourcemanager/ResourceManagerImplTest.java [hunk 5]
```diff
@@ -456,4 +456,42 @@ public void testRuntimeException() {
       assertEquals(exceptionMessage, exception.getCause().getMessage());
     }
   }
+
+  @Test
+  public void testTestOrgPermissions() {
+    ResourceManagerRpcFactory rpcFactoryMock = EasyMock.createMock(ResourceManagerRpcFactory.class);
```
Comment (elharo) [inline src/test/java/com/google/cloud/resourcemanager/ResourceManagerImplTest.java:462]: Prefer Mockito per Google Java policy

**Comment by athakor** [comment]
@elharo all the comment have been addressed PTAL

File: src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/ResourceManagerRpc.java [hunk 6]
```diff
@@ -124,5 +125,13 @@ Integer getInt(Map<Option, ?> options) {
    */
   List<Boolean> testPermissions(String projectId, List<String> permissions);
 
-  // TODO(ajaykannan): implement "Organization" functionality when available (issue #319)
+  /**
+   * Tests whether the caller has the given permissions on the specified Organization. Returns the
+   * permissions and their results corresponding to whether or not the user has the permission in
```
Comment (elharo) [inline src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/ResourceManagerRpc.java:130]: no longer need " corresponding to whether or not the user has the permission in
   * the same position of input list."

Comment (athakor) [reply src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/ResourceManagerRpc.java]: done

File: src/main/java/com/google/cloud/resourcemanager/ResourceManager.java [hunk 7]
```diff
@@ -337,4 +338,20 @@ public static ProjectListOption fields(ProjectField... fields) {
...
+   * @param permissions the set of permissions to check for the resource. Permissions with wildcards
+   *     (such as '*' or 'storage.*') are not allowed.
+   * @return the permissions and their results representing whether the caller has the permissions
+   *     on the specified Organization.
+   * @throws ResourceManagerException upon failure
+   * @see <a href=
+   *     "https://cloud.google.com/resource-manager/reference/rest/v1/organizations/testIamPermissions">
+   *     Resource Manager testIamPermissions</a>
+   */
+  Map<String, Boolean> testOrgPermissions(String resource, List<String> permissions);
```
Comment (elharo) [inline src/main/java/com/google/cloud/resourcemanager/ResourceManager.java:356]: Not new in this PR, and shouldn't be fixed now, but the distinction between ResourceManager and ResourceManagerImpl is likely unnecessary and just makes the code more complicated for no particular reason.

**Review by elharo — APPROVED** [review]

**Pull request merged**